docs/source/development/style-guide.rst

```diff
@@ -0,0 +1,391 @@
+`caret notation <https://python-poetry.org/docs/dependency-specification/#caret-requirements>`_ to allow for minor
+version updates. For example, if you have tested the code using numpy version ``1.24.2``, the best way to specify the
+dependency is to allow for updates to the right most version number:
+
+::
+
+    numpy = "^1.24"
+
+This will allow for patches for security reasons without upgrading to ``1.25`` or beyond. Use your best judgement for
+when packages can upgrade automatically, but try and avoid specifying only one specific version unless it is absolutely
+required.
+
+The ``poetry.lock`` file contains the existing dependencies for the project. These are the dependencies that you should
+install to ensure you're getting the accepted versions for every package. If the ``poetry.lock`` file is not up-to-date,
+there is an automatic process within the repo to update ``poetry.lock`` to match ``pyproject.toml``. To resolve any
+dependency issues before the pre-commit hook, users should run ``poetry lock`` and resolve any dependency problems ahead
+of time. The ``poetry.lock`` file should be part of the files committed in this case.
+
+
+.. _security:
+Security
+--------
+
+The following items should never be committed in the ``imap_processing`` source code or GitHub issues/pull requests:
+
+* Account credentials of any kind (e.g. database usernames/passwords, AWS credentials, etc.)
+* Internal directory structures or filepaths
+* Machine names
+* Proprietary data
+
+If ``imap_processing`` code needs access to this information, it should be stored in a configuration file that is not
+part of the repository.
+
+
+.. _naming-conventions:
+Naming Conventions
+------------------
+
+The ``imap_processing`` repo shall adhere to specific naming conventions to allow for more consistency:
+
+* Repositories shall have branches named ``main`` and ``dev``, instead of ``master`` and ``development``
+* Repositories shall be named with underscores instead of hyphens (e.g. ``imap_processing`` instead of
+  ``imap-processing``)
+
+
+.. _tools-and-library-recommendations:
+Tools and Library Recommendations
+---------------------------------
+
+* ``black`` for checking python code formatting
+* ``poetry`` for dependency management and virtual environments
+* ``pytest`` for unit testing
+* ``ruff`` for checking docstring formatting
+* ``spiceypy`` for using SPICE kernels
+* ``space-packet-parser`` to unpack CCSDS packets
+
+
+.. _versioning:
+Versioning
+----------
+
+Any changes pushed to the ``main`` branch should be tagged with a version number. The version number convention is
+``x.y.z``, where
+
+::
+
+    x = The main version number.  Increase when making incompatible API changes.
+    y = The feature number.  Increase when change contains a new feature with or without bug fixes.
+    z = The hotfix number. Increase when change only contains bug fixes.
+
+
+.. _release-workflow:
+Release Workflow
```

Comment:
This is a real nitpick, but I'd like to have the release workflow in a different file/page, since it isn't really related to the style guide.

Reply:
Agreed. I plan a (near) future PR that will break up the style guide into separate smaller pages, and I can move this section from out under the style guide umbrella.